src/node_quic_socket.h

```diff
@@ -250,7 +250,7 @@ class QuicSocket : public AsyncWrap,
   // value reaches the value of max_connections_per_host_,
   // attempts to create new connections will be ignored
   // until the value falls back below the limit.
-  std::unordered_map<const sockaddr*, size_t, SocketAddress::Hash,
+  std::unordered_map<const sockaddr, size_t, SocketAddress::Hash,
```

Comment:
Unfortunately, this doesn’t quite work, because now we store a key of type `sockaddr`; but `sockaddr_in` and `sockaddr_in6` are “subclasses” of `sockaddr`, and so a key that is just `sockaddr` won’t be able to actually hold the full address.

`sockaddr_storage` would probably the best key type, although that comes with a bit of overhead as it’s supposed to also be able to hold UNIX socket paths which are much longer than IPv4/IPv6 addresses… I’d still use it, and I’m not sure whether QUIC over UNIX sockets is technically impossible or something we should restrict?

Reply:
Thanks for the comment! I would like to give `sockaddr_storage` a try.